pyomo/contrib/mindtpy/nlp_solve.py

```diff
@@ -95,7 +97,9 @@ def solve_subproblem(solve_data, config):
     # Solve the NLP
     nlpopt = SolverFactory(config.nlp_solver)
     nlp_args = dict(config.nlp_solver_args)
-    set_solver_options(nlpopt, solve_data, config, solver_type='nlp')
+    # TODO: Can we move set_solver_options outside of this function?
+    # if not, we can define this function as a method
```

Comment:
I don't have any strong feelings for one or the other. Wha it's your preference?

Reply:
Since set_solver_options is used in single tree implementation, we should keep it as a function.